### PR TITLE
KEYCLOAK-6757 Update MicrosoftIdentityProvider to use the Microsoft G…

### PR DESCRIPTION
**Pull request by sguilhen**
…raph API

Related to https://github.com/keycloak/keycloak/pull/5636

### CODE REVIEW

**Review by matthewhelmke — APPROVED** [review]
Thank you very much. This looks good to me, pending a second review from someone in engineering.

**Review by abstractj — APPROVED** [review]

**Pull request merged**
